sw/device/sca/lib/aes.c

```diff
@@ -0,0 +1,224 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+/**
+ * NOTE: The only intended use of this code is to serve as a PRNG for generating
+ * input data for SCA experiments and penetration testing.
+ * The library is not hardened against any type of attacks, and it should not be
+ * used for any purpose other than stated.
+ *
+ * Implementation of round-functions is based on a transposed-state technique
+ * for 32-bit architecture presented in:
+ *
+ * [1] Bertoni et. al., Efficient Software Implementation of AES on 32-Bit
+ *     Platforms, CHES 2002.
+ *
+ *     https://link.springer.com/content/pdf/10.1007/3-540-36400-5_13.pdf
+ *
```

Comment:
Nice comment header!

Reply:
Maybe it's worth adding that we verify all this in our SCA evaluation setup by doing the same encryptions in Python on the host to detect if host and target ever get out of sync.

Reply:
OK. I've added a line describing this.

Reply:
Perfect, thanks!